webapp/lib/controller.dart

```diff
@@ -90,6 +102,61 @@ void initUI() {
   );
 }
 
+void setupProjectTimers() {
+  var allProjects = view.contentView.projectSelectorView.allProjects;
+  for (var project in allProjects) {
+    var timer = new Timer(Duration(hours: 1), () => view.contentView.stale = true);
+    projectTimers.add(new ProjectTimer(project, timer));
+  }
+}
+
+void setupProjectTimer(String project) {
+  var timer = new Timer(Duration(hours: 1), () => view.contentView.stale = true);
+  projectTimers.add(new ProjectTimer(project, timer));
+}
+
+void checkNeedsReplyMetricsStale([bool isProjectSelection = false]) {
+  List<model.NeedsReplyData> sortedNeedsReplyDataList = List.from(needsReplyDataList);
+  sortedNeedsReplyDataList.sort((d1, d2) => d1.datetime.compareTo(d2.datetime));
+  var allProjects = view.contentView.projectSelectorView.allProjects;
+
+  List<model.NeedsReplyData> latestDataPerProject = [];
+  for (var project in allProjects) {
+    var latestProject = sortedNeedsReplyDataList.where((entry) => entry.project == project).last;
+    latestDataPerProject.add(latestProject);
+  }
+
+  if (projectTimers.length < 1) {
+      setupProjectTimers();
+  } else {
+    for (var entry in latestDataPerProject) {
+      var projectTimer = projectTimers.firstWhere((projectTimer) => projectTimer.project == entry.project);
```

Comment:
It might make sense to use the `orElse` optional for `firstWhere` to return `null` if the project isn't found (e.g. a new project has been set up while the page is open).

Reply:
Noted